src/components/cryptoDollar/cryptoDollarReducer.js

```diff
@@ -0,0 +1,235 @@
+let initialState = {
+  contractState: {
+    loading: true,
+    error: null,
+    data: null
+  },
+  buy: {
+    loading: false,
+    error: null,
+    receipt: null,
+    txHash: ''
+  },
+  sell: {
+    loading: false,
+    error: null,
+    receipt: null,
+    txHash: ''
+  },
+  sellUnpegged: {
+    loading: false,
+    error: null,
+    receipt: null,
+    txHash: ''
+  },
+  transfer: {
+    loading: false,
+    error: null,
+    receipt: null,
+    txHash: ''
+  }
+}
+
+export default function (state = initialState, action) {
+  switch (action.type) {
+    case 'FETCHING_CRYPTODOLLAR_STATE':
+      return {
+        ...state,
+        contractState: {
+          loading: true,
+          data: null
```

Comment:
Don't forget to include spreads on these inner objects as well, for example `...state.contractState`. Otherwise you lose some of the object fields.